### PR TITLE
[Banner] Fix broken Banner snapshot tests

### PR DESCRIPTION
**Pull request by fluxminds**
Banner's snapshot tests were broken caused by https://github.com/material-components/material-components-ios/pull/6991.

This PR fixes the snapshot tests by adding new API support (e.g. layoutMargins), uses layoutMargins to set the banner view's margin and updates the golden images.

### CODE REVIEW

File: components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m [hunk 1]
```diff
@@ -36,9 +37,13 @@ - (void)setUp {
 
   // Uncomment below to recreate all the goldens (or add the following line to the specific
   // test you wish to recreate the golden for).
-  //  self.recordMode = YES;
+//    self.recordMode = YES;
```
Comment (romoore) [inline components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m]: Please revert this change - the older formatting was correct.

File: components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m [hunk 2]
```diff
@@ -36,9 +37,13 @@ - (void)setUp {
...
 
-  self.bannerView = [[MDCBannerView alloc] initWithFrame:CGRectMake(0, 0, 350, 51)];
+  self.bannerView = [[MDCBannerView alloc] initWithFrame:CGRectZero];
+  UIEdgeInsets margins = UIEdgeInsetsZero;
```
Comment (romoore) [inline components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m]: These look like more than "fixing" the tests - can you please fix the PR description?

**Review by romoore — DISMISSED** [review]

File: components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m (hunk 1 above)
Comment (fluxminds) [reply components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m]: Addressed in https://github.com/material-components/material-components-ios/pull/7018/commits/6db8cc73f124506de79933b0abd74dbdec9ac48b.

File: components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m (hunk 2 above)
Comment (fluxminds) [reply components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m]: Sure. Updated the description.

Comment (romoore) [reply components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m]: The description is still remarking on only API changes and golden updates. This appears to be code changes.

Comment (fluxminds) [reply components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m]: Updated with "uses layoutMargins to set the banner view's margin" to make it clear. The code changes here were caused by the API support changes too.

File: components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m [hunk 3]
```diff
@@ -38,7 +39,11 @@ - (void)setUp {
   // test you wish to recreate the golden for).
   //  self.recordMode = YES;
 
-  self.bannerView = [[MDCBannerView alloc] initWithFrame:CGRectMake(0, 0, 350, 51)];
+  self.bannerView = [[MDCBannerView alloc] initWithFrame:CGRectZero];
+  UIEdgeInsets margins = UIEdgeInsetsZero;
+  margins.left = kBannerContentPadding;
+  margins.right = kBannerContentPadding;
+  self.bannerView.layoutMargins = margins;
```
Comment (romoore) [inline components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m]: Please set `directionalLayoutMargins` on iOS 11+.

**Review by romoore — APPROVED** [review]
Approve pending that you update to use `directionLayoutMargins` on supported versions.

File: components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m (hunk 3 above)
Comment (fluxminds) [reply components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m]: Addressed in 0306a767c94b9d09ea2ecae308fe68d7417b5289.

Does Snapshot test also run on iOS 11+ now?

Comment (romoore) [reply components/Banner/examples/tests/snapshot/MDCBannerSnapshotTests.m]: Our snapshot tests have always run on iOS 11.2
